### PR TITLE
[Rust] Add concept headings to introduction.md

### PR DESCRIPTION
**Pull request by ErikSchierboom**
In issue https://github.com/exercism/v3/issues/2293, some changes were introduced.
One of these changes is that the introduction.md should have a separate section for each of the concepts that is listed in the exercise's `concepts` key in the track's `config.json` file.
These section should use a second-tier (`##`) and match the concept casing as specified in the `config.json`.

Take the following `config.json` file:

```json
{
...
"exercises": {
    "concept": [
      {
        "slug": "lucians-luscious-lasagna",
        "name": "Lucian's Luscious Lasagna",
        "uuid": "0e1dd0cf-cdd7-48a9-861e-7253b5d3cabd",
        "concepts": ["basics"],
        "prerequisites": []
      },
      {
        "slug": "bird-watcher",
        "name": "Bird Watcher",
        "uuid": "b6c532c9-1e89-4fbf-8f08-27f5befb5bb8",
        "concepts": ["arrays", "for-loops", "foreach-loops"],
        "prerequisites": ["booleans", "classes", "if-statements"]
      },
      ...
    ],
    ...
},
```

This means that the `introduction.md` of the `lucians-luscious-lasagna` should have one `## basics` heading, whereas the `bird-watcher` exercise's `introduction.md` should have `## arrays`, `## for-loops` and `## foreach-loops` headings. It remains true (and important!) that the introduction concept only shows the minimum necessary information about a concept to allow a student to complete the exercise.

It is important to note that the content that is following these headings is describing the concept in such a way that it can be displayed stand-alone. The text should thus not contain things like "in the previous section on subject X", etc.
This could mean that in some cases (where there are multiple concepts), there could be some work to make the concept introductions be stand-alone.

This PR has been automatically created. It is up to the track maintainers to review the contents and do any potential rewriting. Thanks in advance!

### CODE REVIEW

File: languages/rust/exercises/concept/enums/.docs/introduction.md [hunk 1]
```diff
@@ -1 +1,3 @@
+## enums-basic
```
Comment (coriolinus) [inline languages/rust/exercises/concept/enums/.docs/introduction.md:1]: @ErikSchierboom Are these headings presented as-is to the student, or are they parsed/prettified in any way? `enums-basic` is a good slug for a concept describing basic enums, but it's a pretty bad description to present to a human student.

File: languages/rust/exercises/concept/numbers/.docs/introduction.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## numbers
+## integers
+## floating-point-numbers
```
Comment (coriolinus) [inline languages/rust/exercises/concept/numbers/.docs/introduction.md:3]: We need to rewrite this introduction to split these sections appropriately.

File: languages/rust/exercises/concept/enums/.docs/introduction.md (hunk 1 above)
Comment (ErikSchierboom) [reply languages/rust/exercises/concept/enums/.docs/introduction.md:1]: They will definitely be prettified. The most likely action is to replace them with the corresponding concept's `name` entry as defined in the `config.json` file.

**Comment by efx** [comment]
Apologies on the noise. I am going to decompose the changes in this PR to other ones.

**Pull request closed without merging**